### PR TITLE
Fix service monitor selector

### PR DESCRIPTION
**Pull request by victorboissiere**
Signed-off-by: Victor Boissiere <victor.boissiere@qonto.com>

**Issue #, if available:**

**Description of changes:**
Fix issue on service monitor selector.

Current service monitor selector:
```yaml
  selector:
    matchLabels:
      app.kubernetes.io/component: deployment
      app.kubernetes.io/instance: aws-node-termination-handler
      app.kubernetes.io/name: termination-handler
     ...
```

But the servicelabel  is
```yaml
apiVersion: v1
kind: Service
metadata:
  labels:
    app.kubernetes.io/component: aws-node-termination-handler <== here
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: termination-handler
    ...
```

Fix the issue by ensuring service monitor is target the same labels as the service

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by brycahta** [comment]
@victorboissiere thanks for the PR. Don't be concerned with the failing test as I don't think it relates to your changes. I'll get a PR out this afternoon to fix it; similar change here: https://github.com/aws/amazon-ec2-metadata-mock/pull/168

**Comment by victorboissiere** [comment]
ok @brycahta, let me know if I can do something on my side

File: config/helm/aws-node-termination-handler/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 name: aws-node-termination-handler
 description: A Helm chart for the AWS Node Termination Handler.
 type: application
-version: 0.18.0
+version: 0.18.1
```
Comment (brycahta) [inline config/helm/aws-node-termination-handler/Chart.yaml:5]: Can you remove this update? We increment version when we release (see file's history for more context).

**Review by brycahta — CHANGES_REQUESTED** [review]
Hi @victorboissiere, I got the fix in: https://github.com/aws/aws-node-termination-handler/pull/605. 

You'll just need to rebase so tests pass

File: config/helm/aws-node-termination-handler/Chart.yaml (hunk 1 above)
Comment (victorboissiere) [reply config/helm/aws-node-termination-handler/Chart.yaml]: done

**Review by brycahta — APPROVED** [review]
/lgtm

**Comment by stevehipwell** [comment]
@victorboissiere @brycahta @bwagner5 I don't think this is the correct fix. It looks like I used the wrong template function (or something) for the service labels as they are incorrect not the ServiceMonitor labels. Sorry I didn't see this before the release, I'll open a PR to fix.

**Pull request merged**